app/models/shipit/deploy.rb

```diff
@@ -282,8 +283,13 @@ def update_last_deploy_time
       stack.update(last_deployed_at: ended_at)
     end
 
+    def mark_deploy_ref_update
+      @update_ref = true
+    end
+
     def update_latest_deployed_ref
-      stack.update_latest_deployed_ref
+      stack.update_latest_deployed_ref if @update_ref
+      @update_ref = false
```

Comment:
I felt this falsification was needed in case there was a situation where the Deploy was committed again with some update in the same application/object lifetime as when it first transitioned to success. It wasn't immediately clear to me that that couldn't ever happen.

Reply:
A better approach (checking previous_changes) was recommended. Going with that.